### ISSUE TITLE
Messages sometimes stack up at the bottom

### ISSUE THREAD

**Issue by misterff1** [alecxvs/UTChat#20]
When framerate drops (at least then, not sure if there's other times that it happens too), the chat messages won't be pushed upwards to the right position, but hang at the bottom of the chat. Sending a message in chat seems to solve it and correct that by pushing it up to where it should be at that time. It's not a serious issue, but it does look ugly.

**Comment by alecxvs** [comment]
So the messages that get stuck are stuck on top of each other at the bottom, do the existing messages still move up?

**Comment by misterff1** [comment]
No they don't until the stacked messages are moved to their proper position again. Messages by other players solve this after a while and sending one yourself almost always solves it immediately.

**Comment by alecxvs** [comment]
I don't have any way to reproduce this, can you try taking a screenshot and copying the contents of the client console next time this happens?

**Comment by misterff1** [comment]
Yeah it's more likely to happen on a weaker rig (like mine) as it has more framerate inconsistencies (which seems to create it somehow). I'll get a screenie next time it happens

**Comment by misterff1** [comment]
![image6](https://cloud.githubusercontent.com/assets/3277023/7966207/c6369c62-0a22-11e5-9191-ff84c0ebf659.png)
there's one line that stacks up. This time it was just one that didn't move in (you can tell because it is lower than the line of my Chat Extender), but sometimes two or three messages get printed at the exact same spot down there before they start moving in again

**Comment by alecxvs** [comment]
Given that this only happens during low framerates, I assume that the code during the render hook is being terminated in order to let the next frame render in a timely manner. That is the only reason that the motion and fade effects would stop early; if the code is terminated before updating the internal timers. Since it is time based, the framerate should not have any impact on it and even if the game freezes for the entire duration of the effect it should correct on the final frame. Regardless, measures need to be taken to make sure that the final values of the animation are set.

**Comment by alecxvs** [comment]
@misterff1 Here's the file I messaged you about: https://raw.githubusercontent.com/alecxvs/UTChat/b7d8765fd3a0fd11db64f0c8ed1d99c4c73f069b/utchat/client/utchat.lua

Replace your utchat.lua with that, you should get console messages on the client whenever the fade or motion bug occurs. If I'm right about the cause, you may stop getting fade or motion bugs altogether, but you should still get console messages. If that's the case, then we know this is a fix.

**Comment by misterff1** [comment]
I'm afraid it did not solve the issue. It still happens and no messages are sent to the console.

**Comment by alecxvs** [comment]
I accidentally didn't complete the important bit of code, try this one:
https://raw.githubusercontent.com/alecxvs/UTChat/2d3a3a4d0223b2c95e79da2e8dd90d22badca732/utchat/client/utchat.lua

**Comment by misterff1** [comment]
Still the same thing as my previous comment :/

**Comment by alecxvs** [comment]
Just out of wild curiosity, could you go into utchat.lua and find the two instances of {terminate = true} and change them to false?

**Comment by misterff1** [comment]
Just out of wild curiosity, how is it possible that this fixed it? :P

**Comment by alecxvs** [comment]
It's based on the wild chance that the fade and motion issues are actually two separate problems. Unless this fixed the motion bug too, then I'm at a loss for words.

**Comment by misterff1** [comment]
Nope, it does not. was about to tell you that :)

**Comment by alecxvs** [comment]
More specifically, I realized that there's a bug in the fade that causes it to destroy the fade format before setting the fade to the final amount. This wouldn't be a problem with a high framerate because it would be close enough to the end value that you wouldn't notice a difference. I have no idea what's causing the motion bug because it doesn't sound so simple, so that doesn't close this issue.

**Comment by misterff1** [comment]
yeah I see. Is it safe to just do terminate = false though?

**Comment by alecxvs** [comment]
I'm not sure, I don't know why I added that to fade and not to motion. It was probably a quick fix to fade being weird, but I ironed out most of the fade issues by now anyways. It should be fine, but watch for possible performance issues.

**Comment by misterff1** [comment]
yeah I'll monitor that closely. so far it does not seem to drop performance any more that what's already happening because of the broadcasts :P

**Comment by alecxvs** [comment]
I think I see what's causing motion to be buggy, and the verdict is that the code is terrible. I'll probably rewrite the motion format later.

**Comment by misterff1** [comment]
uh oh, we have a problem with this new version of utchat already lol  . people relogged now and suddenly they chatted in normal chat again and this error gets thrown every time someone uses chat:

![image8](https://cloud.githubusercontent.com/assets/3277023/7993390/208d000a-0b06-11e5-88a9-97e6b0d5cdbd.png)

terminate did not cause this. false or true does not make a difference for this.
I returned to the old utchat and just changed the terminate. seems fine now

**Comment by alecxvs** [comment]
~~Can you copy/paste one of the errors here? The message is cut off from that screenshot.~~
Don't do that, create a new issue please

**Comment by misterff1** [comment]
Sorry for the late response. I can do that, but the question is if that utchat.lua text you sent here actually adds anything to the one in the actual dowloadable build. If not, then there is no real issue here as I just reverted to the official one and had no error messages anymore.
